tools/gather-sysinfo/gather-sysinfo.go

```diff
@@ -1,79 +1,108 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"io/ioutil"
-	"log"
 	"os"
 	"path/filepath"
 	"time"
 
 	"github.com/jaypipes/ghw/pkg/snapshot"
+	"github.com/openshift-kni/debug-tools/pkg/knit/cmd"
+	"github.com/spf13/cobra"
 )
 
+type snapshotOptions struct {
+	dumpList  bool
+	output    string
+	rootDir   string
+	sleepTime int
+}
+
 func main() {
-	debug := flag.Bool("debug", false, "enable debug output")
-	output := flag.String("output", "", "path to clone system information into")
-	rootDir := flag.String("root", "", "pseudofs root - use this if running inside a container")
-	dumpList := flag.Bool("dump", false, "just dump the glob list of expected content and exit")
+	root := cmd.NewRootCommand(newSnapshotCommand)
+
+	if err := root.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func newSnapshotCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
+	opts := &snapshotOptions{}
+	snap := &cobra.Command{
+		Use:   "snapshot",
+		Short: "snapshot pseudofilesystems for offline analysis",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return makeSnapshot(cmd, knitOpts, opts, args)
+		},
+		Args: cobra.NoArgs,
+	}
+	snap.Flags().StringVar(&opts.rootDir, "root", "", "pseudofs root - use this if running inside a container")
+	snap.Flags().StringVar(&opts.output, "output", "", "path to clone system information into")
+	snap.Flags().BoolVar(&opts.dumpList, "dump", false, "just dump the glob list of expected content and exit")
 	// use this to debug container behaviour
-	sleepTime := flag.Int("sleep", 0, "amount of seconds to sleep once done, before exit")
-	flag.Parse()
+	snap.Flags().IntVar(&opts.sleepTime, "sleep", 0, "amount of seconds to sleep once done, before exit")
 
+	return snap
+}
+
+func makeSnapshot(cmd *cobra.Command, knitOpts *cmd.KnitOptions, opts *snapshotOptions, args []string) error {
 	// ghw can't handle duplicates in CopyFilesInto, the operation will fail.
 	// Hence we need to make sure we just don't feed duplicates.
 	fileSpecs := dedupExpectedContent(kniExpectedCloneContent(), snapshot.ExpectedCloneContent())
-	if *dumpList {
+	if opts.dumpList {
 		for _, fileSpec := range fileSpecs {
 			fmt.Printf("%s\n", fileSpec)
```

Comment:
to stdout? on purpose?

Reply:
yep. The list of the glob patterns becomes the expected output, hence we send it on stdout.